src/celengine/render.cpp

```diff
@@ -1678,7 +1688,7 @@ void Renderer::draw(const Observer& observer,
 
     renderForegroundAnnotations(FontNormal);
 
-    if (!selectionVisible && (renderFlags & ShowMarkers))
+    if (isSelectionPointerEnabled && !selectionVisible && (renderFlags & ShowMarkers))
```

Comment:
as you're changing this line, update `(renderFlags & ShowMarkers)` to `(renderFlags & ShowMarkers) != 0`